src/main/App.js

```diff
@@ -63,25 +66,117 @@ const reportStatsData = {
   attendees: 435
 };
 
+let db;
+
+const PrivateRoute = ({
+  uuidPresent,
+  component: Component,
+  render: Render,
+  ...rest
+}) => (
+  <Route
+    {...rest}
+    render={props => {
+      console.log(uuidPresent);
+      if (uuidPresent) {
+        if (Component) {
+          return <Component {...props} />;
+        } else if (Render) {
+          return Render(props);
+        } else {
+          return <div>Error</div>;
+        }
+      } else {
+        return <Redirect to="/login" />;
+      }
+    }}
+  />
+);
+
 class App extends Component {
+  constructor() {
+    super();
+    this.state = { uuidPresent: true };
+    //Define indexeddb instance/version
+    db = new Dexie("devoxx-db");
+    db.version(1).stores({ record: "id,uuid" });
+
+    //open connection to indexeddb - display error if connection failed
+    db.open().catch(error => {
+      alert("uuidDb could not be accessed: " + error);
+    });
+
+    this.rootComponent = this.rootComponent.bind(this);
+    this.uuidExists = this.uuidExists.bind(this);
+    this.uuidExists();
+  }
+
+  uuidExists = () => {
+    return new Promise((resolve, reject) => {
+      //open connection to indexeddb - display error if connection failed
+      db.open("devoxx-db").catch(error => {
+        alert("uuidDb could not be accessed: " + error);
+      });
+      db.record &&
+        db.record
+          .get("0")
+          .then(resolution => {
+            if (resolution) {
+              this.setState({ uuidPresent: true });
+              resolve();
+            } else {
+              throw new Error("No UUID");
+            }
+          })
+          .catch(error => {
+            this.setState({ uuidPresent: false });
+            reject(error);
+          });
+    });
+  };
+
+  rootComponent() {
```

Comment:
don't like the naming of this function - not really clear what it does? signInPage would be better